onnxruntime/core/providers/webnn/builders/impl/scatterND_op_builder.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Intel Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/common.h"
+#include "core/providers/shared/utils/utils.h"
+#include "core/providers/webnn/builders/helper.h"
+#include "core/providers/webnn/builders/model_builder.h"
+#include "core/providers/webnn/builders/op_builder_factory.h"
+
+#include "base_op_builder.h"
+
+namespace onnxruntime {
+namespace webnn {
+
+class ScatterNDOpBuilder : public BaseOpBuilder {
+  // Add operator related.
+ private:
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                               const logging::Logger& logger) const override ORT_MUST_USE_RESULT;
+
+  // Operator support related.
+  bool IsOpSupportedImpl(const InitializedTensorSet& /* initializers */, const Node& node,
+                         const WebnnDeviceType /* device_type */, const logging::Logger& logger) const override;
+  bool HasSupportedInputsImpl(const Node& node, const emscripten::val& wnn_limits,
+                              const logging::Logger& logger) const override;
+};
+
+// Add operator related.
+
+Status ScatterNDOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                                                 const logging::Logger& logger) const {
+  const auto& input_defs = node.InputDefs();
+  emscripten::val data = model_builder.GetOperand(input_defs[0]->Name());
+  emscripten::val indices = model_builder.GetOperand(input_defs[1]->Name());
+  emscripten::val updates = model_builder.GetOperand(input_defs[2]->Name());
+  emscripten::val options = emscripten::val::object();
+  options.set("label", node.Name());
+  emscripten::val output =
+      model_builder.GetBuilder().call<emscripten::val>("scatterND", data, indices, updates, options);
+
+  model_builder.AddOperand(node.OutputDefs()[0]->Name(), std::move(output));
+  return Status::OK();
+}
+
+// Operator support related.
+
+bool ScatterNDOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& /* initializers */, const Node& node,
+                                           const WebnnDeviceType /* device_type */,
+                                           const logging::Logger& logger) const {
+  NodeAttrHelper helper(node);
+  if (helper.Get("reduction", "none") != "none") {
+    LOGS(logger, VERBOSE) << "ScatterND: the type of reduction must be none (default)";
```

Comment:
```suggestion
    LOGS(logger, VERBOSE) << "ScatterND: WebNN only supports reduction type none (default)";
```
If the log mentioned *WebNN* in the output spew, indicating the limitation is specific to WebNN, it would be a little clearer (otherwise someone seeing that might assume it's an broader ORT limitation). @honry: Do we generally mention WebNN in these error messages? Should we? (I'm also fine keeping it so if we generally do not)

*Either way, I think we need another merge from main after that merge conflict so I can reapprove it.

Reply:
Added WebNN in the log👌

Reply:
> @Honry: Do we generally mention WebNN in these error messages? Should we? (I'm also fine keeping it so if we generally do not)

No, but it is nice to have. We can make a centralized improvement in the future.